lib/pacemaker/pcmk_sched_bundle.c

```diff
@@ -19,6 +19,67 @@
 #define PE__VARIANT_BUNDLE 1
 #include <lib/pengine/variant.h>
 
+/*!
+ * \internal
+ * \brief Assign a single bundle replica's resources (other than container)
+ *
+ * \param[in,out] replica    Replica to assign
+ * \param[in]     user_data  Preferred node, if any
+ *
+ * \return true (to indicate that any further replicas should be processed)
+ */
+static bool
+assign_replica(pe__bundle_replica_t *replica, void *user_data)
+{
+    pe_node_t *container_host = NULL;
+    const pe_node_t *prefer = user_data;
+    const pe_resource_t *bundle = pe__const_top_resource(replica->container,
+                                                         true);
+
+    if (replica->ip != NULL) {
+        pe_rsc_trace(bundle, "Assigning bundle %s IP %s",
+                     bundle->id, replica->ip->id);
+        replica->ip->cmds->assign(replica->ip, prefer);
+    }
+
+    container_host = replica->container->allocated_to;
+    if (replica->remote != NULL) {
+        if (pe__is_guest_or_remote_node(container_host)) {
+            /* REMOTE_CONTAINER_HACK: "Nested" connection resources must be on
+             * the same host because Pacemaker Remote only supports a single
+             * active connection.
+             */
+            pcmk__new_colocation("child-remote-with-docker-remote", NULL,
+                                 INFINITY, replica->remote,
+                                 container_host->details->remote_rsc, NULL,
+                                 NULL, true, bundle->cluster);
+        }
+        pe_rsc_trace(bundle, "Assigning bundle %s connection %s",
+                     bundle->id, replica->remote->id);
+        replica->remote->cmds->assign(replica->remote, prefer);
+    }
+
+    if (replica->child != NULL) {
+        pe_node_t *node = NULL;
+        GHashTableIter iter;
+
+        g_hash_table_iter_init(&iter, replica->child->allowed_nodes);
+        while (g_hash_table_iter_next(&iter, NULL, (gpointer *) & node)) {
+            if (!pe__same_node(node, replica->node)
```

Comment:
ouch, good catch. not sure why that didn't show up in regression tests; possibly it would only matter if the child instance's migration threshold was reached

Reply:
Maybe. It is surprising, I agree. Previously we checked the threshold only if we **didn't** hit the "not `pe__same_node()`" condition and assign negative `INFINITY` as the score. Now we set the score to positive `INFINITY` if "not `pe__same_node()", regardless of the threshold. You'd think that would cause a test to change.